FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+                {
+                    xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                    if( xAckInfo.callback != NULL )
+                    {
+                        xAckInfo.callback( xAckInfo.pCommandContext );
+                    }
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_SUBACK:
+                xAckInfo = prvPopAck( packetIdentifier );
+
+                if( xAckInfo.packetId == packetIdentifier )
+                {
+                    pxSubscribeInfo = xAckInfo.pCommandContext->pSubscribeInfo;
+
+                    for( i = 0; i < xAckInfo.pCommandContext->subscriptionCount; i++ )
+                    {
+                        LogInfo( ( "Adding subscription to %.*s",
+                                   pxSubscribeInfo[ i ].topicFilterLength,
+                                   pxSubscribeInfo[ i ].pTopicFilter ) );
+                        LogInfo( ( "Filter length: %u", pxSubscribeInfo[ i ].topicFilterLength ) );
+                        prvAddSubscription( pxSubscribeInfo[ i ].pTopicFilter,
+                                            pxSubscribeInfo[ i ].topicFilterLength,
+                                            xAckInfo.pCommandContext->pResponseQueue );
+                    }
+                }
+                else
+                {
+                    xStatus = MQTTBadResponse;
+                }
+
+                xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                if( xAckInfo.callback != NULL )
+                {
+                    xAckInfo.callback( xAckInfo.pCommandContext );
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_UNSUBACK:
+                xAckInfo = prvPopAck( packetIdentifier );
+
+                if( xAckInfo.packetId == packetIdentifier )
+                {
+                    pxSubscribeInfo = xAckInfo.pCommandContext->pSubscribeInfo;
+
+                    for( i = 0; i < xAckInfo.pCommandContext->subscriptionCount; i++ )
+                    {
+                        LogInfo( ( "Removing subscription to %.*s",
+                                   pxSubscribeInfo[ i ].topicFilterLength,
+                                   pxSubscribeInfo[ i ].pTopicFilter ) );
+                        prvRemoveSubscription( pxSubscribeInfo[ i ].pTopicFilter,
+                                               pxSubscribeInfo[ i ].topicFilterLength,
+                                               xAckInfo.pCommandContext->pResponseQueue );
+                    }
+                }
+                else
+                {
+                    xStatus = MQTTBadResponse;
+                }
+
+                xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                if( xAckInfo.callback != NULL )
+                {
+                    xAckInfo.callback( xAckInfo.pCommandContext );
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_PUBREC:
```

Comment:
I think in general, it is better to have more comments in the code to improve the readability of this demo application. Specifically, I think it is better to some comments about why we are not handling these packets.

Reply:
Added comments